src/test/java/org/kohsuke/github/GHPullRequestTest.java

```diff
@@ -421,6 +421,64 @@ public void setLabels() throws Exception {
         assertEquals(label, labels.iterator().next().getName());
     }
 
+    @Test
+    // Requires push access to the test repo to pass
+    public void addLabels() throws Exception {
+        GHPullRequest p = getRepository().createPullRequest("addLabels", "test/stable", "master", "## test");
+        String addedLabel1 = "addLabels_label_name_1";
+        p.setLabels(addedLabel1);
+
+        String addedLabel2 = "addLabels_label_name_2";
+        p.addLabels(addedLabel2);
+
+        Collection<GHLabel> labels = getRepository().getPullRequest(p.getNumber()).getLabels();
+        assertEquals(2, labels.size());
+        assertThat(labels,
+                containsInAnyOrder(hasProperty("name", equalTo(addedLabel1)),
+                        hasProperty("name", equalTo(addedLabel2))));
+    }
+
+    @Test
+    // Requires push access to the test repo to pass
+    public void addLabelsConcurrencyIssue() throws Exception {
+        String addedLabel1 = "addLabelsConcurrencyIssue_label_name_1";
+        String addedLabel2 = "addLabelsConcurrencyIssue_label_name_2";
+
+        GHPullRequest p1 = getRepository()
+                .createPullRequest("addLabelsConcurrencyIssue", "test/stable", "master", "## test");
+        p1.getLabels();
+
+        GHPullRequest p2 = getRepository().getPullRequest(p1.getNumber());
+        p2.addLabels(addedLabel2);
+
+        p1.addLabels(addedLabel1);
```

Comment:
Also do a case where you add two labels in one method call - which should produce one API call, not two.